Makefile

```diff
@@ -5,8 +5,11 @@ MAKEFLAGS += --no-builtin-variables
 COVERAGE_FILE ?= .coverage
 
 # allow overriding the name of the venv directory
-VENV_DIR ?= .venv
-UV_PROJECT_ENVIRONMENT=${VENV_DIR}
+DEFAULT_XDSL_VENV_DIR ?= .venv
+
+# use activated venv if any
+VENV_DIR ?= $(if $(VIRTUAL_ENV),$(VIRTUAL_ENV),$(DEFAULT_XDSL_VENV_DIR))
+export UV_PROJECT_ENVIRONMENT=${VENV_DIR}
```

Comment:
Can we not simplify as

```makefile
# allow overriding the name of the venv directory
VENV_DIR ?= .venv

# use activated venv if any
export UV_PROJECT_ENVIRONMENT=$(if $(VIRTUAL_ENV),$(VIRTUAL_ENV),$(VENV_DIR))
```

Where `VENV_DIR` is just ignored if we are already in an active venv?

Reply:
This keeps the same make usage as we had been using prior to all the changes to accommodate `uv`.
But I don't feel strong about it.

Reply:
I'll try this